src/injector/src/main.rs

```diff
@@ -86,90 +95,82 @@ fn unpack(sha_sum: &String) {
-    let url = request.url();
-    let url_segments: Vec<_> = url.split("/").collect();
-    let url_seg_len = url_segments.len();
-
-    if url_seg_len >= 4 && url_segments[1] == "v2" {
-        let tag_index = url_seg_len - 1;
-        let object_index = url_seg_len - 2;
-
-        let object_type = url_segments[object_index];
-
-        if object_type == "manifests" {
-            let tag_or_digest = url_segments[tag_index].to_owned();
-
-            let namespaced_name = url_segments[2..object_index].join("/");
-
-            // this route handles (GET) (/v2/**/manifests/<tag>)
-            if request.method() == "GET" {
-                return handle_get_manifest(&root, &namespaced_name, &tag_or_digest);
-            // this route handles (HEAD) (/v2/**/manifests/<tag>)
-            } else if request.method() == "HEAD" {
-                // a normal HEAD response has an empty body, but due to rouille not allowing for an override
-                // on Content-Length, we respond the same as a GET
-                return accept!(
-                    request,
-                    OCI_MIME_TYPE => {
-                        handle_get_manifest(&root, &namespaced_name, &tag_or_digest)
-                    },
-                    "*/*" => Response::empty_406()
-                );
-            }
-        // this route handles (GET) (/v2/**/blobs/<digest>)
-        } else if object_type == "blobs" && request.method() == "GET" {
-            let digest = url_segments[tag_index].to_owned();
-            return handle_get_digest(&root, &digest);
-        }
-    }
+async fn handler(Path(path): Path<String>) -> Response {
+    println!("request: {}", path);
+    let path = &path;
+    let manifest = Regex::new("(.+)/manifests/(.+)").unwrap();
+    let blob = Regex::new(".+/([^/]+)").unwrap();
+
+    if manifest.is_match(path){
+        let caps = manifest.captures(path).unwrap();
+        let name = caps.get(1).unwrap().as_str().to_string();
+        let reference = caps.get(2).unwrap().as_str().to_string();
+        handle_get_manifest(name, reference).await
+
 
-    Response::empty_404()
+    }else if blob.is_match(&path) {
+        let caps = blob.captures(path).unwrap();
+        let tag = caps.get(1).unwrap().as_str().to_string();
+        handle_get_digest(tag).await
+    } else {
+        Response::builder()
+        .status(StatusCode::NOT_FOUND)
+        .body(format!("Not Found"))
+        .unwrap()
+        .into_response()
+  }
 }
 
 /// Handles the GET request for the manifest (only returns a OCI manifest regardless of Accept header)
-fn handle_get_manifest(root: &Path, name: &String, tag: &String) -> Response {
-    let index = fs::read_to_string(root.join("index.json")).expect("read index.json");
+async fn handle_get_manifest(name: String, reference: String) -> Response {
+    println!("name {}, reference {}", name, reference);
+    let index = fs::read_to_string(PathBuf::from("/zarf-seed").join("index.json")).expect("read index.json");
```

Comment:
nit: style for `expect` https://doc.rust-lang.org/std/option/enum.Option.html#recommended-message-style